articles/sql-or-nosql-when-to-choose-what/index.md

```diff
@@ -0,0 +1,123 @@
+Relations can exit in different ways such as
+- One to one 
+- One to many
+- Many to many
+
+SQL statements will help you to perform CRUD operations such as CREATE, READ, UPDATE, and DELETE. These operations will help you to retrieve and modify data.
+If we were to have a student info database with table student and course, this is how we can represent a relation in an SQL database.
+
+![image title](\tables.png)
+
+![image title](\sql.png)
+
+SQL is a simple language that is extremely powerful. You can create multiple databases with multiple tables to store data with foreign key mapping. Each row in SQL database has its specific value character such as text and intergers.
+
+![image title](\relatiomal.png)
+
+SQL databases need a server that runs a database management system (DBMS). Some of the common SQL databases include [MySQL](https://www.mysql.com/why-mysql/), [Oracle Database](https://www.oracle.com/database/), [PostgreSQL](https://www.postgresql.org/about/), [Microsoft SQL Server](https://www.microsoft.com/en-gb/sql-server/), and [MariaDB](https://mariadb.org/documentation/).
+
+### NoSQL
+
+NoSQL does not mean “no SQL”. It is a non-SQL database structure with a non-relational mechanism for storing and retrieving data. Its data model does not have tabular schema as used in relational databases. You do not need to create tables (no rows and columns). NoSQL databases store and manage data in ways that allow high operational speed and great flexibility. It is used to build real-time web apps and big data to handle large quantities of unstructured and semi-structured data. They are designed to perform high levels of read and write while scaling horizontally. JSON is the common representation of non-structured data.
+NoSQL example context:
+```sql
+Database --> Collection(s) --> Objects(s)
+```
+If we refer back to SQL, the collection would be the table and the object will be the rows and columns. Just like SQL where a database has several tables, NoSQL creates multiple collections. In this case, there are no relations between the collections. Rows and columns will be the objects of the NoSQL.
+Like SQL, NoSQL has a unique id that uniquely identifies a specific object within a collection. Nested objects and array are supported where an object can be inside another object. If we were to represent the SQL student info database in NoSQL, this how it would look like:
+
+```json
+Student {
```

Comment:
Since you have displayed your data in JSON, every key and value should be enclosed in quotes. For the example you've given,

```json
{
  "Student": {
    "_id": "Objectld('14s5gg98we9d’)",
    "firstName": "Alex",
    "lastName": "Brooks",
    "username": "alex09_b",
    "Courses": "[{name: Web Dev}, {name: React js}]"
  }
}
```